solutions/swb-reference/integration-tests/tests/multiStep/paab.test.ts

```diff
@@ -0,0 +1,452 @@
+
+    console.log('Creating Dataset1 for Project1...');
+    const randomTextGenerator = new RandomTextGenerator(setup.getSettings().get('runId'));
+    const datasetName = randomTextGenerator.getFakeText('paab-test');
+    const dataSetBody = CreateDataSetRequestParser.parse({
+      storageName: setup.getSettings().get('DataSetsBucketName'),
+      awsAccountId: setup.getSettings().get('mainAccountId'),
+      path: datasetName, // using same name to help potential troubleshooting
+      name: datasetName,
+      region: setup.getSettings().get('awsRegion'),
+      owner: getProjectAdminRole(project1Id),
+      ownerType: 'GROUP',
+      type: 'internal',
+      permissions: [
+        {
+          identity: getResearcherRole(project1Id),
+          identityType: 'GROUP',
+          accessLevel: 'read-write'
+        }
+      ]
+    });
+    const { data: ds1 } = await pa1Session.resources.projects
+      .project(project1Id)
+      .dataSets()
+      .create(dataSetBody, false);
+
+    console.log('Verifying PA1 CANNOT see Project2...');
+    // List Projects
+    const { data: listProjects1 }: ListProjectsResponse = await pa1Session.resources.projects.get();
+    expect(listProjects1.data.filter((proj) => proj.id === project2Id).length).toEqual(0);
+
+    // Get Projects
+    try {
+      await pa1Session.resources.projects.project(project2Id).get();
+    } catch (err) {
+      checkHttpError(err, unauthorizedHttpError);
+    }
+
+    console.log('Verify PA1 CANNOT see ETC2...');
+    // Get ETC
+    try {
+      await pa1Session.resources.projects
+        .project(project2Id)
+        .environmentTypes()
+        .environmentType(etId)
+        .configurations()
+        .environmentTypeConfig(etc2.id)
+        .get();
+    } catch (e) {
+      checkHttpError(e, unauthorizedHttpError);
+    }
+    // List ETCs for Project
+    const { data: listEtcs1 }: ListETCsResponse = await pa1Session.resources.projects
+      .project(project1Id)
+      .environmentTypes()
+      .environmentType(etId)
+      .configurations()
+      .get();
+    expect(listEtcs1.data.filter((etc) => etc.id === etc2.id).length).toEqual(0);
+
+    console.log('Verifying PA1 CANNOT create an Environment in Project2...');
+    try {
+      await pa1Session.resources.projects.project(project2Id).environments().create();
+    } catch (e) {
+      checkHttpError(e, unauthorizedHttpError);
+    }
+
+    console.log('Verifying PA2 CANNOT see Project1...');
+    // List Projects
+    const { data: listProjects2 }: ListProjectsResponse = await pa2Session.resources.projects.get();
+    expect(listProjects2.data.filter((proj) => proj.id === project1Id).length).toEqual(0);
+    // Get Projects
+    try {
+      await pa2Session.resources.projects.project(project1Id).get();
+    } catch (err) {
+      checkHttpError(err, unauthorizedHttpError);
+    }
+
+    console.log('Verifying PA2 CANNOT see Dataset1...');
+    // List Datasets for Project
```

Comment:
are we waiting for other implementation to uncomment this?

Reply:
Yeah, @bhudnell is currently working on List Datasets for Project.